serde-jackson/src/test/groovy/io/micronaut/serde/jackson/CustomPrimitiveSerdeSpec.groovy

```diff
@@ -0,0 +1,125 @@
+package io.micronaut.serde.jackson
+
+import io.micronaut.context.annotation.Primary
+import io.micronaut.context.annotation.Secondary
+import io.micronaut.core.annotation.NonNull
+import io.micronaut.core.type.Argument
+import io.micronaut.serde.*
+import io.micronaut.serde.annotation.Serdeable
+import io.micronaut.test.annotation.MockBean
+import io.micronaut.test.extensions.spock.annotation.MicronautTest
+import jakarta.inject.Inject
+import spock.lang.Specification
+
+@MicronautTest
+class CustomPrimitiveSerdeSpec extends Specification {
+
+    @Inject ObjectMapper objectMapper
+
+    void "test custom primitive type encoder"() {
+        expect:
+        objectMapper.writeValueAsString(new TestWithBoxedInt(value: 1)) == '{"value":201}'
+        objectMapper.writeValueAsString(new TestWithPrimitiveInt(value: 1)) == '{"value":201}'
+        objectMapper.readValue('{"value":1}', TestWithBoxedInt).value == 101
+        objectMapper.readValue('{"value":1}', TestWithPrimitiveInt).value == 101
+    }
+
+    void "test custom primitive type encoder when multiple are defined"() {
+        expect:
+        objectMapper.writeValueAsString(new TestWithBoxedLong(value: 1L)) == '{"value":201}'
+        objectMapper.writeValueAsString(new TestWithPrimitiveLong(value: 1L)) == '{"value":201}'
+        objectMapper.readValue('{"value":1}', TestWithBoxedLong).value == 101L
+        objectMapper.readValue('{"value":1}', TestWithPrimitiveLong).value == 101L
+    }
+
+    @Serdeable
+    static class TestWithBoxedInt {
+        Integer value
+    }
+
+    @Serdeable
+    static class TestWithPrimitiveInt {
+        int value
+    }
+
+    @Serdeable
+    static class TestWithBoxedLong {
+        Long value
+    }
+
+    @Serdeable
+    static class TestWithPrimitiveLong {
+        long value
+    }
+
+    @MockBean
+    Serde<Integer> integerSerde() {
```

Comment:
Why `@MockBean`? Just put `@Singleton` or `@Prototype`

Reply:
maybe it is to scope the bean so it is only active for the test?

Reply:
I followed the structure that's present in [serde-jackson/src/test/groovy/io/micronaut/serde/jackson/CustomStringSerdeSpec.groovy](https://github.com/micronaut-projects/micronaut-serialization/blob/master/serde-jackson/src/test/groovy/io/micronaut/serde/jackson/CustomStringSerdeSpec.groovy), my assumption was that this scopes the bean just to the single test class, otherwise some bean conditional logic needs to be introduced

Reply:
I see, we should have something like @TestBean for that

Reply:
Thanks for the PR @oujesky